node-pg-native crashes node

brianc opened the pull request:
I have a tcp socket server which forwards a significant amount of data to postgres. Periodically node-pg-native (I believe) crashes node. Here is the stack trace I get:

```
*** glibc detected *** node: free(): invalid next size (fast): 0x000000000222d110 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x76a16)[0x7fd6cfb84a16]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x6c)[0x7fd6cfb897bc]
node(_ZN2v88internal13GlobalHandles31PostGarbageCollectionProcessingENS0_16GarbageCollectorEPNS0_8GCTracerE+0x14c)[0x82310c]
node(_ZN2v88internal4Heap24PerformGarbageCollectionENS0_16GarbageCollectorEPNS0_8GCTracerENS_15GCCallbackFlagsE+0x250)[0x854040]
node(_ZN2v88internal4Heap14CollectGarbageENS0_16GarbageCollectorEPKcS4_NS_15GCCallbackFlagsE+0x15b)[0x8547ab]
node(_ZN2v88internal7Factory19NewRawOneByteStringEiNS0_13PretenureFlagE+0x7a)[0x7fea8a]
node(_ZN2v88internal6String11SlowFlattenENS0_6HandleINS0_10ConsStringEEENS0_13PretenureFlagE+0x6e)[0x96e6de]
node(_ZN2v88internal6String7FlattenENS0_6HandleIS1_EENS0_13PretenureFlagE+0x98)[0x74f918]
node(_ZNK2v86String9WriteUtf8EPciPii+0x1202)[0x7691f2]
node(_ZN4node11StringBytes5WriteEPN2v87IsolateEPcmNS1_6HandleINS1_5ValueEEENS_8encodingEPi+0x1f8)[0xb292f8]
node(_ZN4node10StreamWrap15WriteStringImplILNS_8encodingE1EEEvRKN2v820FunctionCallbackInfoINS3_5ValueEEE+0x190)[0xb2d050]
[0x2c7b8b611849]
======= Memory map: ========
00400000-00e98000 r-xp 00000000 00:1d 7733                               /usr/local/bin/node
01098000-010af000 rwxp 00a98000 00:1d 7733                               /usr/local/bin/node
010af000-010b9000 rwxp 00000000 00:00 0 
02161000-0249d000 rwxp 00000000 00:00 0                                  [heap]
56f05100000-56f05200000 rwxp 00000000 00:00 0 
c308bf00000-c308c000000 rwxp 00000000 00:00 0 
dbef1600000-dbef1700000 rwxp 00000000 00:00 0 
dbfa1a00000-dbfa1b00000 rwxp 00000000 00:00 0 
f2c59e00000-f2c59f00000 rwxp 00000000 00:00 0 
10c721000000-10c721100000 rwxp 00000000 00:00 0 
12a3281af000-12a3281b0000 r-xp 00000000 00:00 0 
137a69700000-137a69725000 rwxp 00000000 00:00 0 
15a102000000-15a102015000 rwxp 00000000 00:00 0 
17195e600000-17195e700000 rwxp 00000000 00:00 0 
1783cfe00000-1783cff00000 rwxp 00000000 00:00 0 
17a4e3100000-17a4e3200000 rwxp 00000000 00:00 0 
180b05900000-180b05a00000 rwxp 00000000 00:00 0 
1899c4800000-1899c48c5000 rwxp 00000000 00:00 0 
1b4064e00000-1b4064f00000 rwxp 00000000 00:00 0 
1e2089600000-1e2089625000 rwxp 00000000 00:00 0 
21a64b700000-21a64b800000 rwxp 00000000 00:00 0 
21f5f7400000-21f5f7500000 rwxp 00000000 00:00 0 
22fa9037e000-22fa903fe000 rwxp 00000000 00:00 0 
22fa903fe000-22fa9057e000 ---p 00000000 00:00 0 
236933600000-236933700000 rwxp 00000000 00:00 0 
2389f6300000-2389f6400000 rwxp 00000000 00:00 0 
23a70fb00000-23a70fc00000 rwxp 00000000 00:00 0 
2408f5000000-2408f5100000 rwxp 00000000 00:00 0 
2825da700000-2825da800000 rwxp 00000000 00:00 0 
293c28f00000-293c29000000 rwxp 00000000 00:00 0 
2b0777000000-2b0777100000 rwxp 00000000 00:00 0 
2c712a000000-2c712a100000 rwxp 00000000 00:00 0 
2c7b8b09a000-2c7b8b100000 ---p 00000000 00:00 0 
2c7b8b100000-2c7b8b105000 rwxp 00000000 00:00 0 
2c7b8b105000-2c7b8b106000 ---p 00000000 00:00 0 
2c7b8b106000-2c7b8b107000 rwxp 00000000 00:00 0 
2c7b8b107000-2c7b8b200000 ---p 00000000 00:00 0 
2c7b8b200000-2c7b8b205000 rwxp 00000000 00:00 0 
2c7b8b205000-2c7b8b206000 ---p 00000000 00:00 0 
2c7b8b206000-2c7b8b207000 rwxp 00000000 00:00 0 
2c7b8b207000-2c7b8b300000 ---p 00000000 00:00 0 
2c7b8b300000-2c7b8b305000 rwxp 00000000 00:00 0 
2c7b8b305000-2c7b8b306000 ---p 00000000 00:00 0 
2c7b8b306000-2c7b8b307000 rwxp 00000000 00:00 0 
2c7b8b307000-2c7b8b400000 ---p 00000000 00:00 0 
2c7b8b400000-2c7b8b405000 rwxp 00000000 00:00 0 
2c7b8b405000-2c7b8b406000 ---p 00000000 00:00 0 
2c7b8b406000-2c7b8b4ff000 rwxp 00000000 00:00 0 
2c7b8b4ff000-2c7b8b500000 ---p 00000000 00:00 0 
2c7b8b500000-2c7b8b505000 rwxp 00000000 00:00 0 
2c7b8b505000-2c7b8b506000 ---p 00000000 00:00 0 
2c7b8b506000-2c7b8b5ff000 rwxp 00000000 00:00 0 
2c7b8b5ff000-2c7b8b600000 ---p 00000000 00:00 0 
2c7b8b600000-2c7b8b605000 rwxp 00000000 00:00 0 
2c7b8b605000-2c7b8b606000 ---p 00000000 00:00 0 
2c7b8b606000-2c7b8b6ff000 rwxp 00000000 00:00 0 
2c7b8b6ff000-2c7b8b700000 ---p 00000000 00:00 0 
2c7b8b700000-2c7b8b705000 rwxp 00000000 00:00 0 
2c7b8b705000-2c7b8b706000 ---p 00000000 00:00 0 
2c7b8b706000-2c7b8b7ff000 rwxp 00000000 00:00 0 
2c7b8b7ff000-2c7b8b800000 ---p 00000000 00:00 0 
2c7b8b800000-2c7b8b805000 rwxp 00000000 00:00 0 
2c7b8b805000-2c7b8b806000 ---p 00000000 00:00 0 
2c7b8b806000-2c7b8b8ff000 rwxp 00000000 00:00 0 
2c7b8b8ff000-2c7b8b900000 ---p 00000000 00:00 0 
2c7b8b900000-2c7b8b905000 rwxp 00000000 00:00 0 
2c7b8b905000-2c7b8b906000 ---p 00000000 00:00 0 
2c7b8b906000-2c7b8b9ff000 rwxp 00000000 00:00 0 
2c7b8b9ff000-2c7b8ba00000 ---p 00000000 00:00 0 
2c7b8ba00000-2c7b8ba05000 rwxp 00000000 00:00 0 
2c7b8ba05000-2c7b8ba06000 ---p 00000000 00:00 0 
2c7b8ba06000-2c7b8baff000 rwxp 00000000 00:00 0 
2c7b8baff000-2c7b8bb00000 ---p 00000000 00:00 0 
2c7b8bb00000-2c7b8bb05000 rwxp 00000000 00:00 0 
2c7b8bb05000-2c7b8bb06000 ---p 00000000 00:00 0 
2c7b8bb06000-2c7b8bbff000 rwxp 00000000 00:00 0 
2c7b8bbff000-2c7b8bc00000 ---p 00000000 00:00 0 
2c7b8bc00000-2c7b8bc05000 rwxp 00000000 00:00 0 
2c7b8bc05000-2c7b8bc06000 ---p 00000000 00:00 0 
2c7b8bc06000-2c7b8bcff000 rwxp 00000000 00:00 0 
2c7b8bcff000-2c7b8bd00000 ---p 00000000 00:00 0 
2c7b8bd00000-2c7b8bd05000 rwxp 00000000 00:00 0 
2c7b8bd05000-2c7b8bd06000 ---p 00000000 00:00 0 
2c7b8bd06000-2c7b8bdff000 rwxp 00000000 00:00 0 
2c7b8bdff000-2c7b8be00000 ---p 00000000 00:00 0 
2c7b8be00000-2c7b8be05000 rwxp 00000000 00:00 0 
2c7b8be05000-2c7b8be06000 ---p 00000000 00:00 0 
2c7b8be06000-2c7b8beff000 rwxp 00000000 00:00 0 
2c7b8beff000-2c7b8bf00000 ---p 00000000 00:00 0 
2c7b8bf00000-2c7b8bf05000 rwxp 00000000 00:00 0 
2c7b8bf05000-2c7b8bf06000 ---p 00000000 00:00 0 
2c7b8bf06000-2c7b8bfff000 rwxp 00000000 00:00 0 
2c7b8bfff000-2c7b8c500000 ---p 00000000 00:00 0 
2c7b8c500000-2c7b8c505000 rwxp 00000000 00:00 0 
2c7b8c505000-2c7b8c506000 ---p 00000000 00:00 0 
2c7b8c506000-2c7b8c5ff000 rwxp 00000000 00:00 0 
2c7b8c5ff000-2c7bab09a000 ---p 00000000 00:00 0 
2c9713600000-2c9713700000 rwxp 00000000 00:00 0 
3153512f2000-315351300000 ---p 00000000 00:00 0 
315351300000-315351320000 rwxp 00000000 00:00 0 
315351320000-315351352000 ---p 00000000 00:00 0 
321afba00000-321afba35000 rwxp 00000000 00:00 0 
32b477100000-32b477200000 rwxp 00000000 00:00 0 
33332d100000-33332d200000 rwxp 00000000 00:00 0 
349d2254e000-349d2294e000 rwxp 00000000 00:00 0 
34f56f800000-34f56f900000 rwxp 00000000 00:00 0 
3643ac200000-3643ac300000 rwxp 00000000 00:00 0 
376952b00000-376952c00000 rwxp 00000000 00:00 0 
3777b9800000-3777b9900000 rwxp 00000000 00:00 0 
3d239e000000-3d239f000000 rwxp 00000000 00:00 0 
3d239f000000-3d23a0000000 rwxp 00000000 00:00 0 
3d688a900000-3d688aa00000 rwxp 00000000 00:00 0 
3fd2a9000000-3fd2a9100000 rwxp 00000000 00:00 0 
7fd6c5ffc000-7fd6c5ffd000 ---p 00000000 00:00 0 
7fd6c5ffd000-7fd6c67fd000 rwxp 00000000 00:00 0                          [stack:25307]
7fd6c67fd000-7fd6c67fe000 ---p 00000000 00:00 0 
7fd6c67fe000-7fd6c6ffe000 rwxp 00000000 00:00 0                          [stack:25306]
7fd6c6ffe000-7fd6c6fff000 ---p 00000000 00:00 0 
7fd6c6fff000-7fd6c77ff000 rwxp 00000000 00:00 0                          [stack:25305]
7fd6c77ff000-7fd6c7800000 ---p 00000000 00:00 0 
7fd6c7800000-7fd6c8000000 rwxp 00000000 00:00 0                          [stack:25304]
7fd6c8000000-7fd6c810a000 rwxp 00000000 00:00 0 
7fd6c810a000-7fd6cc000000 ---p 00000000 00:00 0 
7fd6cc520000-7fd6cc523000 r-xp 00000000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7fd6cc523000-7fd6cc722000 ---p 00003000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7fd6cc722000-7fd6cc723000 rwxp 00002000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7fd6cc723000-7fd6cc734000 r-xp 00000000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7fd6cc734000-7fd6cc933000 ---p 00011000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7fd6cc933000-7fd6cc934000 r-xp 00010000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7fd6cc934000-7fd6cc935000 rwxp 00011000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7fd6cc935000-7fd6cc945000 r-xp 00000000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7fd6cc945000-7fd6ccb44000 ---p 00010000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7fd6ccb44000-7fd6ccb45000 r-xp 0000f000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7fd6ccb45000-7fd6ccb46000 rwxp 00010000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7fd6ccb46000-7fd6ccbc1000 r-xp 00000000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0
7fd6ccbc1000-7fd6ccdc1000 ---p 0007b000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0
7fd6ccdc1000-7fd6ccdc2000 r-xp 0007b000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0
7fd6ccdc2000-7fd6ccdc5000 rwxp 0007c000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0
7fd6ccdc5000-7fd6cce7e000 r-xp 00000000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7fd6cce7e000-7fd6cd07d000 ---p 000b9000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7fd6cd07d000-7fd6cd083000 r-xp 000b8000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7fd6cd083000-7fd6cd085000 rwxp 000be000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7fd6cd085000-7fd6cd09f000 r-xp 00000000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7fd6cd09f000-7fd6cd29e000 ---p 0001a000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7fd6cd29e000-7fd6cd29f000 r-xp 00019000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7fd6cd29f000-7fd6cd2a0000 rwxp 0001a000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7fd6cd2a0000-7fd6cd2ae000 r-xp 00000000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7fd6cd2ae000-7fd6cd4ad000 ---p 0000e000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7fd6cd4ad000-7fd6cd4ae000 r-xp 0000d000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7fd6cd4ae000-7fd6cd4af000 rwxp 0000e000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7fd6cd4af000-7fd6cd4c2000 r-xp 00000000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7fd6cd4c2000-7fd6cd6c1000 ---p 00013000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7fd6cd6c1000-7fd6cd6c2000 r-xp 00012000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7fd6cd6c2000-7fd6cd6c3000 rwxp 00013000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7fd6cd6c3000-7fd6cd6c5000 rwxp 00000000 00:00 0 
7fd6cd6c5000-7fd6cd6c8000 r-xp 00000000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7fd6cd6c8000-7fd6cd8c7000 ---p 00003000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7fd6cd8c7000-7fd6cd8c8000 r-xp 00002000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7fd6cd8c8000-7fd6cd8c9000 rwxp 00003000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7fd6cd8c9000-7fd6cd8d1000 r-xp 00000000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7fd6cd8d1000-7fd6cdad0000 ---p 00008000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7fd6cdad0000-7fd6cdad1000 r-xp 00007000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7fd6cdad1000-7fd6cdad2000 rwxp 00008000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7fd6cdad2000-7fd6cdaf8000 r-xp 00000000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7fd6cdaf8000-7fd6cdcf8000 ---p 00026000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7fd6cdcf8000-7fd6cdcf9000 r-xp 00026000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7fd6cdcf9000-7fd6cdcfa000 rwxp 00027000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7fd6cdcfa000-7fd6cdcfb000 rwxp 00000000 00:00 0 
7fd6cdcfb000-7fd6cdd11000 r-xp 00000000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7fd6cdd11000-7fd6cdf10000 ---p 00016000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7fd6cdf10000-7fd6cdf11000 r-xp 00015000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7fd6cdf11000-7fd6cdf12000 rwxp 00016000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7fd6cdf12000-7fd6cdf5e000 r-xp 00000000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7fd6cdf5e000-7fd6ce15e000 ---p 0004c000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7fd6ce15e000-7fd6ce160000 r-xp 0004c000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7fd6ce160000-7fd6ce161000 rwxp 0004e000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7fd6ce161000-7fd6ce163000 rwxp 00000000 00:00 0 
7fd6ce163000-7fd6ce19f000 r-xp 00000000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7fd6ce19f000-7fd6ce39f000 ---p 0003c000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7fd6ce39f000-7fd6ce3a0000 r-xp 0003c000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7fd6ce3a0000-7fd6ce3a2000 rwxp 0003d000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7fd6ce3a2000-7fd6ce3a5000 r-xp 00000000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7fd6ce3a5000-7fd6ce5a4000 ---p 00003000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7fd6ce5a4000-7fd6ce5a5000 r-xp 00002000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7fd6ce5a5000-7fd6ce5a6000 rwxp 00003000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7fd6ce5a6000-7fd6ce66f000 r-xp 00000000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7fd6ce66f000-7fd6ce86e000 ---p 000c9000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7fd6ce86e000-7fd6ce878000 r-xp 000c8000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7fd6ce878000-7fd6ce87a000 rwxp 000d2000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7fd6ce87a000-7fd6cea44000 r-xp 00000000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fd6cea44000-7fd6cec44000 ---p 001ca000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fd6cec44000-7fd6cec5f000 r-xp 001ca000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fd6cec5f000-7fd6cec6e000 rwxp 001e5000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fd6cec6e000-7fd6cec72000 rwxp 00000000 00:00 0 
7fd6cec72000-7fd6cecc8000 r-xp 00000000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
7fd6cecc8000-7fd6ceec8000 ---p 00056000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
7fd6ceec8000-7fd6ceecb000 r-xp 00056000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
7fd6ceecb000-7fd6ceed2000 rwxp 00059000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0
7fd6ceed2000-7fd6ceefc000 r-xp 00000000 00:1d 14648                      /usr/lib/libpq.so.5.4
7fd6ceefc000-7fd6cf0fb000 ---p 0002a000 00:1d 14648                      /usr/lib/libpq.so.5.4
7fd6cf0fb000-7fd6cf0fd000 r-xp 00029000 00:1d 14648                      /usr/lib/libpq.so.5.4
7fd6cf0fd000-7fd6cf0ff000 rwxp 0002b000 00:1d 14648                      /usr/lib/libpq.so.5.4
7fd6cf0ff000-7fd6cf10d000 r-xp 00000000 00:1d 23420                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7fd6cf10d000-7fd6cf30c000 ---p 0000e000 00:1d 23420                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7fd6cf30c000-7fd6cf30d000 rwxp 0000d000 00:1d 23420                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7fd6cf30d000-7fd6cf30e000 ---p 00000000 00:00 0 
7fd6cf30e000-7fd6cfb0e000 rwxp 00000000 00:00 0                          [stack:24007]
7fd6cfb0e000-7fd6cfc90000 r-xp 00000000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7fd6cfc90000-7fd6cfe90000 ---p 00182000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7fd6cfe90000-7fd6cfe94000 r-xp 00182000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7fd6cfe94000-7fd6cfe95000 rwxp 00186000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7fd6cfe95000-7fd6cfe9a000 rwxp 00000000 00:00 0 
7fd6cfe9a000-7fd6cfeb1000 r-xp 00000000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7fd6cfeb1000-7fd6d00b0000 ---p 00017000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7fd6d00b0000-7fd6d00b1000 r-xp 00016000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7fd6d00b1000-7fd6d00b2000 rwxp 00017000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7fd6d00b2000-7fd6d00b6000 rwxp 00000000 00:00 0 
7fd6d00b6000-7fd6d00cb000 r-xp 00000000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7fd6d00cb000-7fd6d02cb000 ---p 00015000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7fd6d02cb000-7fd6d02cc000 rwxp 00015000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7fd6d02cc000-7fd6d034d000 r-xp 00000000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7fd6d034d000-7fd6d054c000 ---p 00081000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7fd6d054c000-7fd6d054d000 r-xp 00080000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7fd6d054d000-7fd6d054e000 rwxp 00081000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7fd6d054e000-7fd6d0636000 r-xp 00000000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7fd6d0636000-7fd6d0836000 ---p 000e8000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7fd6d0836000-7fd6d083e000 r-xp 000e8000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7fd6d083e000-7fd6d0840000 rwxp 000f0000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7fd6d0840000-7fd6d0855000 rwxp 00000000 00:00 0 
7fd6d0855000-7fd6d0857000 r-xp 00000000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7fd6d0857000-7fd6d0a57000 ---p 00002000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7fd6d0a57000-7fd6d0a58000 r-xp 00002000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7fd6d0a58000-7fd6d0a59000 rwxp 00003000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7fd6d0a59000-7fd6d0a60000 r-xp 00000000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7fd6d0a60000-7fd6d0c5f000 ---p 00007000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7fd6d0c5f000-7fd6d0c60000 r-xp 00006000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7fd6d0c60000-7fd6d0c61000 rwxp 00007000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7fd6d0c61000-7fd6d0c81000 r-xp 00000000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7fd6d0e64000-7fd6d0e65000 ---p 00000000 00:00 0 
7fd6d0e65000-7fd6d0e7b000 rwxp 00000000 00:00 0                          [stack:24008]
7fd6d0e7e000-7fd6d0e80000 rwxp 00000000 00:00 0 
7fd6d0e80000-7fd6d0e81000 r-xp 0001f000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7fd6d0e81000-7fd6d0e82000 rwxp 00020000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7fd6d0e82000-7fd6d0e83000 rwxp 00000000 00:00 0 
7fff9e42c000-7fff9e44d000 rwxp 00000000 00:00 0                          [stack]
7fff9e5d3000-7fff9e5d5000 r--p 00000000 00:00 0                          [vvar]
7fff9e5d5000-7fff9e5d7000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
```

EliSnow commented:
Node has been very stable; no more crashes. I think you nailed it!

brianc commented:
:dancer: Thank you __so much__ for your help in reporting this, testing it multiple times, and being patient waiting for the fix!! :clap: 

Going to release a new version to npm now with the fixes.

EliSnow commented:
I hate to do this, but I realized a few days ago that I hadn't really tested your patch. I updated my package.json to pull down pg-native but I forgot to change `require("pg")` to `require("pg").native`. So I remedied that problem and sure enough there was another crash. Here's the error dump I have in the logs:

```
*** glibc detected *** node: double free or corruption (out): 0x00000000026c37b0 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x76a16)[0x7f0401fc4a16]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x6c)[0x7f0401fc97bc]
node(_ZN2v88internal13GlobalHandles31PostGarbageCollectionProcessingENS0_16GarbageCollectorEPNS0_8GCTracerE+0x3a8)[0x823368]
node(_ZN2v88internal4Heap24PerformGarbageCollectionENS0_16GarbageCollectorEPNS0_8GCTracerENS_15GCCallbackFlagsE+0x250)[0x854040]
node(_ZN2v88internal4Heap14CollectGarbageENS0_16GarbageCollectorEPKcS4_NS_15GCCallbackFlagsE+0x15b)[0x8547ab]
node(_ZN2v88internal4Heap17CollectAllGarbageEiPKcNS_15GCCallbackFlagsE+0x75)[0x854c55]
node(_ZN2v88internal9Execution25HandleStackGuardInterruptEPNS0_7IsolateE+0x432)[0x7f3d22]
[0x21fa747060a2]
======= Memory map: ========
00400000-00e98000 r-xp 00000000 00:1d 7733                               /usr/local/bin/node
01098000-010af000 rwxp 00a98000 00:1d 7733                               /usr/local/bin/node
010af000-010b9000 rwxp 00000000 00:00 0
02647000-02b07000 rwxp 00000000 00:00 0                                  [heap]
b95dc00000-b95dd00000 rwxp 00000000 00:00 0
218a8a00000-218a8b00000 rwxp 00000000 00:00 0
31b62100000-31b62200000 rwxp 00000000 00:00 0
4fb81500000-4fb81600000 rwxp 00000000 00:00 0
5979c100000-5979c200000 rwxp 00000000 00:00 0
5f2e3600000-5f2e3625000 rwxp 00000000 00:00 0
61f17700000-61f17800000 rwxp 00000000 00:00 0
9037e000000-9037e100000 rwxp 00000000 00:00 0
a38d0200000-a38d0235000 rwxp 00000000 00:00 0
12aacac00000-12aacac15000 rwxp 00000000 00:00 0
14f5fa700000-14f5fa725000 rwxp 00000000 00:00 0
162203800000-162203900000 rwxp 00000000 00:00 0
1738d3894000-1738d38c0000 ---p 00000000 00:00 0
1738d38c0000-1738d38e0000 rwxp 00000000 00:00 0
1738d38e0000-1738d38f4000 ---p 00000000 00:00 0
1866bde58000-1866be258000 rwxp 00000000 00:00 0
18c58a5ae000-18c58a5af000 r-xp 00000000 00:00 0
1b4a74000000-1b4a74100000 rwxp 00000000 00:00 0
1c328e400000-1c328e500000 rwxp 00000000 00:00 0
1c6564000000-1c6565000000 rwxp 00000000 00:00 0
1c6565000000-1c6566000000 rwxp 00000000 00:00 0
1f021ec00000-1f021ed00000 rwxp 00000000 00:00 0
217df3100000-217df3200000 rwxp 00000000 00:00 0
21fa743b2000-21fa74400000 ---p 00000000 00:00 0
21fa74400000-21fa74405000 rwxp 00000000 00:00 0
21fa74405000-21fa74406000 ---p 00000000 00:00 0
21fa74406000-21fa74407000 rwxp 00000000 00:00 0
21fa74407000-21fa74500000 ---p 00000000 00:00 0
21fa74500000-21fa74505000 rwxp 00000000 00:00 0
21fa74505000-21fa74506000 ---p 00000000 00:00 0
21fa74506000-21fa74507000 rwxp 00000000 00:00 0
21fa74507000-21fa74600000 ---p 00000000 00:00 0
21fa74600000-21fa74605000 rwxp 00000000 00:00 0
21fa74605000-21fa74606000 ---p 00000000 00:00 0
21fa74606000-21fa74607000 rwxp 00000000 00:00 0
21fa74607000-21fa74700000 ---p 00000000 00:00 0
21fa74700000-21fa74705000 rwxp 00000000 00:00 0
21fa74705000-21fa74706000 ---p 00000000 00:00 0
21fa74706000-21fa747ff000 rwxp 00000000 00:00 0
21fa747ff000-21fa74800000 ---p 00000000 00:00 0
21fa74800000-21fa74805000 rwxp 00000000 00:00 0
21fa74805000-21fa74806000 ---p 00000000 00:00 0
21fa74806000-21fa748ff000 rwxp 00000000 00:00 0
21fa748ff000-21fa74900000 ---p 00000000 00:00 0
21fa74900000-21fa74905000 rwxp 00000000 00:00 0
21fa74905000-21fa74906000 ---p 00000000 00:00 0
21fa74906000-21fa749ff000 rwxp 00000000 00:00 0
21fa749ff000-21fa943b2000 ---p 00000000 00:00 0
24a4af500000-24a4af600000 rwxp 00000000 00:00 0
275d6c36e000-275d6c3ee000 rwxp 00000000 00:00 0
275d6c3ee000-275d6c56e000 ---p 00000000 00:00 0
281d1a600000-281d1a700000 rwxp 00000000 00:00 0
2b031e000000-2b031e100000 rwxp 00000000 00:00 0
2b6a6e600000-2b6a6e700000 rwxp 00000000 00:00 0
2c0a91000000-2c0a91100000 rwxp 00000000 00:00 0
2dce4c600000-2dce4c700000 rwxp 00000000 00:00 0
2dfc33f00000-2dfc34000000 rwxp 00000000 00:00 0
2e31f4900000-2e31f4a00000 rwxp 00000000 00:00 0
2e3473600000-2e3473700000 rwxp 00000000 00:00 0
2ee63ce00000-2ee63cf00000 rwxp 00000000 00:00 0
33f76aa00000-33f76ab00000 rwxp 00000000 00:00 0
34baa2800000-34baa28c5000 rwxp 00000000 00:00 0
392ba5b00000-392ba5c00000 rwxp 00000000 00:00 0
3a2520700000-3a2520800000 rwxp 00000000 00:00 0
3aaa03500000-3aaa03600000 rwxp 00000000 00:00 0
3b5581f00000-3b5582000000 rwxp 00000000 00:00 0
3e336b700000-3e336b800000 rwxp 00000000 00:00 0
3fb9d6200000-3fb9d6300000 rwxp 00000000 00:00 0
3fc462900000-3fc462a00000 rwxp 00000000 00:00 0
7f03f8000000-7f03f8269000 rwxp 00000000 00:00 0
7f03f8269000-7f03fc000000 ---p 00000000 00:00 0
7f03fc95c000-7f03fc95d000 ---p 00000000 00:00 0
7f03fc95d000-7f03fd15d000 rwxp 00000000 00:00 0                          [stack:7803]
7f03fd15d000-7f03fd15e000 ---p 00000000 00:00 0
7f03fd15e000-7f03fd95e000 rwxp 00000000 00:00 0                          [stack:7802]
7f03fd95e000-7f03fd95f000 ---p 00000000 00:00 0
7f03fd95f000-7f03fe15f000 rwxp 00000000 00:00 0                          [stack:7801]
7f03fe15f000-7f03fe160000 ---p 00000000 00:00 0
7f03fe160000-7f03fe960000 rwxp 00000000 00:00 0                          [stack:7800]
7f03fe960000-7f03fe963000 r-xp 00000000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7f03fe963000-7f03feb62000 ---p 00003000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7f03feb62000-7f03feb63000 rwxp 00002000 00:1d 7672                       /lib/x86_64-linux-gnu/libgpg-error.so.0.8.0
7f03feb63000-7f03feb74000 r-xp 00000000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7f03feb74000-7f03fed73000 ---p 00011000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7f03fed73000-7f03fed74000 r-xp 00010000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7f03fed74000-7f03fed75000 rwxp 00011000 00:1d 7713                       /usr/lib/x86_64-linux-gnu/libp11-kit.so.0.0.0
7f03fed75000-7f03fed85000 r-xp 00000000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7f03fed85000-7f03fef84000 ---p 00010000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7f03fef84000-7f03fef85000 r-xp 0000f000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7f03fef85000-7f03fef86000 rwxp 00010000 00:1d 7694                       /usr/lib/x86_64-linux-gnu/libtasn1.so.3.1.16
7f03fef86000-7f03ff001000 r-xp 00000000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0 7f03ff001000-7f03ff201000 ---p 0007b000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0 7f03ff201000-7f03ff202000 r-xp 0007b000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0 7f03ff202000-7f03ff205000 rwxp 0007c000 00:1d 7671                       /lib/x86_64-linux-gnu/libgcrypt.so.11.7.0 7f03ff205000-7f03ff2be000 r-xp 00000000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7f03ff2be000-7f03ff4bd000 ---p 000b9000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7f03ff4bd000-7f03ff4c3000 r-xp 000b8000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7f03ff4c3000-7f03ff4c5000 rwxp 000be000 00:1d 7717                       /usr/lib/x86_64-linux-gnu/libgnutls.so.26.22.4
7f03ff4c5000-7f03ff4df000 r-xp 00000000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7f03ff4df000-7f03ff6de000 ---p 0001a000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7f03ff6de000-7f03ff6df000 r-xp 00019000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7f03ff6df000-7f03ff6e0000 rwxp 0001a000 00:1d 7722                       /usr/lib/x86_64-linux-gnu/libsasl2.so.2.0.25
7f03ff6e0000-7f03ff6ee000 r-xp 00000000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7f03ff6ee000-7f03ff8ed000 ---p 0000e000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7f03ff8ed000-7f03ff8ee000 r-xp 0000d000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7f03ff8ee000-7f03ff8ef000 rwxp 0000e000 00:1d 7680                       /usr/lib/x86_64-linux-gnu/liblber-2.4.so.2.8.3
7f03ff8ef000-7f03ff902000 r-xp 00000000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7f03ff902000-7f03ffb01000 ---p 00013000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7f03ffb01000-7f03ffb02000 r-xp 00012000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7f03ffb02000-7f03ffb03000 rwxp 00013000 00:1d 749                        /lib/x86_64-linux-gnu/libresolv-2.13.so
7f03ffb03000-7f03ffb05000 rwxp 00000000 00:00 0
7f03ffb05000-7f03ffb08000 r-xp 00000000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7f03ffb08000-7f03ffd07000 ---p 00003000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7f03ffd07000-7f03ffd08000 r-xp 00002000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7f03ffd08000-7f03ffd09000 rwxp 00003000 00:1d 7668                       /lib/x86_64-linux-gnu/libkeyutils.so.1.4
7f03ffd09000-7f03ffd11000 r-xp 00000000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f03ffd11000-7f03fff10000 ---p 00008000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f03fff10000-7f03fff11000 r-xp 00007000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f03fff11000-7f03fff12000 rwxp 00008000 00:1d 7677                       /usr/lib/x86_64-linux-gnu/libkrb5support.so.0.1
7f03fff12000-7f03fff38000 r-xp 00000000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f03fff38000-7f0400138000 ---p 00026000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f0400138000-7f0400139000 r-xp 00026000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f0400139000-7f040013a000 rwxp 00027000 00:1d 7687                       /usr/lib/x86_64-linux-gnu/libk5crypto.so.3.1
7f040013a000-7f040013b000 rwxp 00000000 00:00 0
7f040013b000-7f0400151000 r-xp 00000000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7f0400151000-7f0400350000 ---p 00016000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7f0400350000-7f0400351000 r-xp 00015000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7f0400351000-7f0400352000 rwxp 00016000 00:1d 770                        /lib/x86_64-linux-gnu/libz.so.1.2.7
7f0400352000-7f040039e000 r-xp 00000000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7f040039e000-7f040059e000 ---p 0004c000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7f040059e000-7f04005a0000 r-xp 0004c000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7f04005a0000-7f04005a1000 rwxp 0004e000 00:1d 7685                       /usr/lib/x86_64-linux-gnu/libldap_r-2.4.so.2.8.3
7f04005a1000-7f04005a3000 rwxp 00000000 00:00 0
7f04005a3000-7f04005df000 r-xp 00000000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7f04005df000-7f04007df000 ---p 0003c000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7f04007df000-7f04007e0000 r-xp 0003c000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7f04007e0000-7f04007e2000 rwxp 0003d000 00:1d 7681                       /usr/lib/x86_64-linux-gnu/libgssapi_krb5.so.2.2
7f04007e2000-7f04007e5000 r-xp 00000000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f04007e5000-7f04009e4000 ---p 00003000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f04009e4000-7f04009e5000 r-xp 00002000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f04009e5000-7f04009e6000 rwxp 00003000 00:1d 703                        /lib/x86_64-linux-gnu/libcom_err.so.2.1
7f04009e6000-7f0400aaf000 r-xp 00000000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f0400aaf000-7f0400cae000 ---p 000c9000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f0400cae000-7f0400cb8000 r-xp 000c8000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f0400cb8000-7f0400cba000 rwxp 000d2000 00:1d 7714                       /usr/lib/x86_64-linux-gnu/libkrb5.so.3.3
7f0400cba000-7f0400e84000 r-xp 00000000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7f0400e84000-7f0401084000 ---p 001ca000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7f0401084000-7f040109f000 r-xp 001ca000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7f040109f000-7f04010ae000 rwxp 001e5000 00:1d 7676                       /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7f04010ae000-7f04010b2000 rwxp 00000000 00:00 0
7f04010b2000-7f0401108000 r-xp 00000000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 7f0401108000-7f0401308000 ---p 00056000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 7f0401308000-7f040130b000 r-xp 00056000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 7f040130b000-7f0401312000 rwxp 00059000 00:1d 7718                       /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 7f0401312000-7f040133c000 r-xp 00000000 00:1d 12409                      /usr/lib/libpq.so.5.4
7f040133c000-7f040153b000 ---p 0002a000 00:1d 12409                      /usr/lib/libpq.so.5.4
7f040153b000-7f040153d000 r-xp 00029000 00:1d 12409                      /usr/lib/libpq.so.5.4
7f040153d000-7f040153f000 rwxp 0002b000 00:1d 12409                      /usr/lib/libpq.so.5.4
7f040153f000-7f040154d000 r-xp 00000000 00:1d 26700                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7f040154d000-7f040174c000 ---p 0000e000 00:1d 26700                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7f040174c000-7f040174d000 rwxp 0000d000 00:1d 26700                      /var/lib/app/node_modules/pg-native/node_modules/libpq/build/Release/addon.node
7f040174d000-7f040174e000 ---p 00000000 00:00 0
7f040174e000-7f0401f4e000 rwxp 00000000 00:00 0                          [stack:8947]
7f0401f4e000-7f04020d0000 r-xp 00000000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7f04020d0000-7f04022d0000 ---p 00182000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7f04022d0000-7f04022d4000 r-xp 00182000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7f04022d4000-7f04022d5000 rwxp 00186000 00:1d 698                        /lib/x86_64-linux-gnu/libc-2.13.so
7f04022d5000-7f04022da000 rwxp 00000000 00:00 0
7f04022da000-7f04022f1000 r-xp 00000000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7f04022f1000-7f04024f0000 ---p 00017000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7f04024f0000-7f04024f1000 r-xp 00016000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7f04024f1000-7f04024f2000 rwxp 00017000 00:1d 745                        /lib/x86_64-linux-gnu/libpthread-2.13.so
7f04024f2000-7f04024f6000 rwxp 00000000 00:00 0
7f04024f6000-7f040250b000 r-xp 00000000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7f040250b000-7f040270b000 ---p 00015000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7f040270b000-7f040270c000 rwxp 00015000 00:1d 712                        /lib/x86_64-linux-gnu/libgcc_s.so.1
7f040270c000-7f040278d000 r-xp 00000000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7f040278d000-7f040298c000 ---p 00081000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7f040298c000-7f040298d000 r-xp 00080000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7f040298d000-7f040298e000 rwxp 00081000 00:1d 717                        /lib/x86_64-linux-gnu/libm-2.13.so
7f040298e000-7f0402a76000 r-xp 00000000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f0402a76000-7f0402c76000 ---p 000e8000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f0402c76000-7f0402c7e000 r-xp 000e8000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f0402c7e000-7f0402c80000 rwxp 000f0000 00:1d 1561                       /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.17
7f0402c80000-7f0402c95000 rwxp 00000000 00:00 0
7f0402c95000-7f0402c97000 r-xp 00000000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7f0402c97000-7f0402e97000 ---p 00002000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7f0402e97000-7f0402e98000 r-xp 00002000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7f0402e98000-7f0402e99000 rwxp 00003000 00:1d 706                        /lib/x86_64-linux-gnu/libdl-2.13.so
7f0402e99000-7f0402ea0000 r-xp 00000000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7f0402ea0000-7f040309f000 ---p 00007000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7f040309f000-7f04030a0000 r-xp 00006000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7f04030a0000-7f04030a1000 rwxp 00007000 00:1d 751                        /lib/x86_64-linux-gnu/librt-2.13.so
7f04030a1000-7f04030c1000 r-xp 00000000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7f04032a4000-7f04032a5000 ---p 00000000 00:00 0
7f04032a5000-7f04032bb000 rwxp 00000000 00:00 0                          [stack:8948]
7f04032be000-7f04032c0000 rwxp 00000000 00:00 0
7f04032c0000-7f04032c1000 r-xp 0001f000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7f04032c1000-7f04032c2000 rwxp 00020000 00:1d 682                        /lib/x86_64-linux-gnu/ld-2.13.so
7f04032c2000-7f04032c3000 rwxp 00000000 00:00 0
7fff6846a000-7fff6848b000 rwxp 00000000 00:00 0                          [stack]
7fff684f6000-7fff684f8000 r--p 00000000 00:00 0                          [vvar]
7fff684f8000-7fff684fa000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
```
Sorry for the false hope earlier.

I am more than happy to continue testing out potential fixes, but if you're fed up with trying to figure out this issue, the "pg" package works fine for my needs and I can use that.

EliSnow commented:
Just a thought, sorry if this is naive I don't know c++, is it possible you could get a build of libpq to me which provides better stack traces and might give us more clues when it crashes?

brianc commented:
Hey @EliSnow - I worked _more_ on this sucker and eventually put in a fix.  I isolated the crash. I found a few places where double-frees _might_ happen and removed them. Eventually I adjusted how the GC collects the object & everything should be in order now.  I know you probably happy using the pure javascript version now (it's really not far off in speed) but if you ever come back & try the native again I hope everything works for you!  Again, I appreciate you being so patient through all this. It's people like you that make doing OSS so fun!

EliSnow commented:
Glad I can help. Is the new fix what's currently in master?

brianc commented:
Yah! `npm i pg-native` should do the trick. You'll want to be using
`>=libpq@1.5.1`

On Wed, Jan 21, 2015 at 2:21 PM, EliSnow <notifications@github.com> wrote:

> Glad I can help. Is the new fix what's currently in master?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brianc/node-pg-native/pull/11#issuecomment-70903011>.
>

EliSnow commented:
I will give it a try next week. This week's pretty busy

EliSnow commented:
I apologize for not getting to this sooner. I have tested this out over the last few days and it appears to be fully functional. Great work!

brianc commented:
No worries! Thanks so much for helping!

On Fri, Feb 20, 2015 at 12:32 PM, EliSnow <notifications@github.com> wrote:

> I apologize for not getting to this sooner. I have tested this out over
> the last few days and it appears to be fully functional. Great work!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brianc/node-pg-native/pull/11#issuecomment-75268031>.
>
